File: HackerRank/Sparse-Arrays.py
def matchingStrings(stringList, queries):
  result = []
  for i in range(len(queries)):
    result.append(stringList.count(queries[i]))
  return result


print(matchingStrings(["aba", "baba", "aba", "xzxb"], ["aba", "xzxb", "ab"]))

File: HackerRank/JavaScript/Sparse-Arrays.js
function matchingStrings(stringList, queries) {
  // Write your code here
  console.log(stringList, queries);
  const result = new Array(queries.length);
  for (let i = 0; i < queries.length; i++) {
    let counter = 0;
    for (let j = 0; j < stringList.length; j++) {
      console.log(stringList[j], queries[i], stringList[j] == queries[i]);
      if (stringList[j] === queries[i]) counter++;
    }
    result[i] = counter;
  }
  return result;
}

console.log(
  matchingStrings(["aba", "baba", "aba", "xzxb"], ["aba", "xzxb", "ab"])
);
